components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/api/client/GradleEnterpriseApiClient.java

```diff
@@ -37,33 +38,59 @@ public class GradleEnterpriseApiClient {
     private final ConsoleLogger logger;
 
     public GradleEnterpriseApiClient(URL baseUrl, CustomValueNames customValueNames, ConsoleLogger logger) {
+        this.logger = logger;
+        this.baseUrl = baseUrl;
         this.customValueNames = customValueNames;
+
         ApiClient client = new ApiClient();
-        client.setHttpClient(configureProxyAuthentication(client.getHttpClient()));
+        client.setHttpClient(configureHttpClient(client.getHttpClient()));
         client.setBasePath(baseUrl.toString());
         AuthenticationConfigurator.configureAuth(baseUrl, client, logger);
 
-        this.baseUrl = baseUrl;
         this.apiClient = new GradleEnterpriseApi(client);
-        this.logger = logger;
     }
 
-    private OkHttpClient configureProxyAuthentication(OkHttpClient httpClient) {
-        return httpClient.newBuilder()
-                .proxyAuthenticator((route, response) -> {
-                    if (response.code() == 407) {
-                        String scheme = response.request().url().scheme().toLowerCase(Locale.ROOT);
-                        String proxyUser = System.getProperty(scheme + ".proxyUser");
-                        String proxyPassword = System.getProperty(scheme + ".proxyPassword");
-                        if (proxyUser != null && proxyPassword != null) {
-                            return response.request().newBuilder()
-                                    .header("Proxy-Authorization", Credentials.basic(proxyUser, proxyPassword))
-                                    .build();
-                        }
+    private OkHttpClient configureHttpClient(OkHttpClient httpClient) {
+        OkHttpClient.Builder httpClientBuilder = httpClient.newBuilder();
+
+        configureSsl(httpClientBuilder);
+        configureProxyAuthentication(httpClientBuilder);
+
+        return httpClientBuilder.build();
+    }
+
+    private void configureSsl(OkHttpClient.Builder httpClientBuilder) {
+        HandshakeCertificates.Builder trustedCertsBuilder = new HandshakeCertificates.Builder()
+            .addPlatformTrustedCertificates();
+
+        if (allowUntrustedServer()) {
+            trustedCertsBuilder.addInsecureHost(baseUrl.getHost());
+            httpClientBuilder.hostnameVerifier((hostname, session) -> baseUrl.getHost().equals(hostname));
+        }
+
+        HandshakeCertificates trustedCerts = trustedCertsBuilder.build();
+        httpClientBuilder.sslSocketFactory(trustedCerts.sslSocketFactory(), trustedCerts.trustManager());
+    }
+
+    private boolean allowUntrustedServer() {
+        return Boolean.parseBoolean(System.getProperty("allowUntrustedServer"));
+    }
+
+    private void configureProxyAuthentication(OkHttpClient.Builder httpClientBuilder) {
+        httpClientBuilder
+            .proxyAuthenticator((route, response) -> {
+                if (response.code() == 407) {
+                    String scheme = response.request().url().scheme().toLowerCase(Locale.ROOT);
```

Comment:
Is Local.ROOT env specific? If so, I would use a locale that is fixex and matches what a scheme can be encoded in. (ASCII?)

Reply:
Locale.ROOT is not env specific. It is the base Locale from which all other Locale's inherit from. Locale.ROOT is considered the best locale to use for locale insensitive strings (such as the URL scheme used here, which should always be lowercased the same way regardless of the user's locale).

Reply:
Interesting.